discovery/gossiper.go

```diff
@@ -631,33 +633,62 @@ func (d *AuthenticatedGossiper) syncBlockHeight() {
 	}
 }
 
+// cachedFutureMsg is a future message that's saved to the `futureMsgs` cache.
+type cachedFutureMsg struct {
+	// msg is the network message.
+	msg *networkMsg
+
+	// height is the block height.
+	height uint32
+}
+
+// Size returns the size of the message.
+func (c *cachedFutureMsg) Size() (uint64, error) {
+	// Return a constant 1.
+	return 1, nil
+}
+
 // resendFutureMessages takes a block height, resends all the future messages
-// found at that height and deletes those messages found in the gossiper's
-// futureMsgs.
+// found below and equal to that height and deletes those messages found in the
+// gossiper's futureMsgs.
 func (d *AuthenticatedGossiper) resendFutureMessages(height uint32) {
-	result, err := d.futureMsgs.Get(height)
+	// msgs are the target messages.
+	msgs := make([]*networkMsg, 0)
```

Comment:
Can use `var` declaration if not pre-sizing any of the elements below.

Reply:
Done.